### PR TITLE
Adds ToMut trait.

### PR DESCRIPTION
**Pull request by remexre**
Basically the same as ToRef, except with `&mut` instead of `&`.

### CODE REVIEW

File: core/src/coproduct.rs [hunk 1]
```diff
@@ -476,6 +476,30 @@ impl<Head, Tail> Coproduct<Head, Tail> {
         ToRef::to_ref(self)
     }
 
+    /// Borrow each variant of the Coproduct mutably.
```
Comment (Centril) [inline core/src/coproduct.rs]:
```suggestion
    /// Borrow each variant of the `Coproduct` mutably.
```

File: core/src/hlist.rs [hunk 2]
```diff
@@ -324,6 +324,25 @@ macro_rules! gen_inherent_methods {
                 ToRef::to_ref(self)
             }
 
+            /// Return an HList where the contents are mutable references
```
Comment (Centril) [inline core/src/hlist.rs]:
```suggestion
            /// Return an `HList` where the contents are mutable references
```

File: core/src/hlist.rs [hunk 3]
```diff
@@ -324,6 +324,25 @@ macro_rules! gen_inherent_methods {
...
+            /// to the original HList on which this method was called.
```
Comment (Centril) [inline core/src/hlist.rs]:
```suggestion
            /// to the original `HList` on which this method was called.
```

File: core/src/hlist.rs [hunk 4]
```diff
@@ -324,6 +324,25 @@ macro_rules! gen_inherent_methods {
...
+            /// ```
+            /// # #[macro_use] extern crate frunk; fn main() {
+            /// assert_eq!(hlist![].to_mut(), hlist![]);
+            ///
+            /// assert_eq!(hlist![1, true].to_mut(), hlist![&mut 1, &mut true]);
+            /// # }
+            /// ```
+            #[inline(always)]
+            pub fn to_mut<'a>(&'a mut self) -> <Self as ToMut<'a>>::Output
+                where Self: ToMut<'a>,
```
Comment (Centril) [inline core/src/hlist.rs]: Inconsistent style as compared to other places in this PR; use the style you've used before.

File: core/src/traits.rs [hunk 5]
```diff
@@ -18,6 +18,17 @@ pub trait ToRef<'a> {
     fn to_ref(&'a self) -> Self::Output;
 }
 
+/// An alternative to AsMut that does not force the reference type to be a pointer itself.
```
Comment (Centril) [inline core/src/traits.rs]:
```suggestion
/// An alternative to `AsMut` that does not force the reference type to be a pointer itself.
```

File: core/src/traits.rs [hunk 6]
```diff
@@ -18,6 +18,17 @@ pub trait ToRef<'a> {
...
+///
+/// This parallels [ToRef]; see it for more information.
```
Comment (Centril) [inline core/src/traits.rs]:
```suggestion
/// This parallels [`ToRef`]; see it for more information.
```

File: core/src/traits.rs [hunk 7]
```diff
@@ -18,6 +18,17 @@ pub trait ToRef<'a> {
...
+///
+/// [ToRef]: trait.ToRef.html
```
Comment (Centril) [inline core/src/traits.rs]:
```suggestion
/// [`ToRef`]: trait.ToRef.html
```

**Comment by remexre** [comment]
The style's largely a result of copy-pasting the equivalent ToRef parts; should I change those too?

**Comment by Centril** [comment]
@remexre I think we can do that as a follow up rustfmt-only PR for cleaner diffs or as part of the release procedure.

**Comment by remexre** [comment]
Looks like it's stalling out on nightly on Travis; works on my machine (on nightly) though.

**Comment by Centril** [comment]
@lloydmeta Feels generally easiest to keep track of things if each PR adds to Unreleased.

**Comment by lloydmeta** [comment]
@remexre would you mind adding a quick summary of this change to [the Changelog](https://github.com/lloydmeta/frunk/blob/master/CHANGELOG.md)?

**Comment by remexre** [comment]
Should be done.

**Pull request merged**